### ISSUE TITLE
dotnet watch script injection still doesn't update etag

### ISSUE THREAD

**Issue by SteveSandersonMS** [dotnet/aspnetcore#36219]
I reported this [before](https://github.com/dotnet/aspnetcore/issues/27548) and it was closed as done, but the problem still seems to exist in current RC2 builds.

Repro:

 * In a Blazor WebAssembly hosted project, `cd Server` then `dotnet watch`
 * After the site has appeared in your browser, `ctrl+c` in the command prompt
 * Now `dotnet run` (so we're running *without* watch now)
 * Reload the page in the same browser tab you had before

Expected:

 * Page loads and works as normal for `dotnet run`

Actual:

![image](https://user-images.githubusercontent.com/1101362/132325070-53d13d79-9c21-4c85-ad52-4e8536813b46.png)

This is because, even though the browser is requesting `/` with an `if-none-match: <etag>`, the server is still lying and saying `304 Not Modified` when the page content *would* be modified if you did a hard-refresh. In my case the server returns the etag `1d7a3cb24b130e3` both with and without the injected `<script src="/_framework/aspnetcore-browser-refresh.js"></script>`, so the caching behavior will go wrong.

The inverse problem also occurs if you try to switch from `dotnet run` to `dotnet watch` - until you hard reload, it continues to use the same `/` content as from `dotnet run`, so you don't get hot reload.

So in summary, it doesn't work to switch in either direction between `watch` and `run`, which will certainly be a common situation.

## We knew this would happen

@pranavkm commented about this in https://github.com/dotnet/aspnetcore/pull/27778:

> While this issue may crop up if you alternate between dotnet run and dotnet watch run but we haven't seen this being an issue as yet.

I even replied *I like this approach!* so it's clear I hadn't fully internalised what level of strangeness would ensue, which I should have done. Sorry. Maybe I thought "we haven't seen this being an issue" as meaning "it's a theoretical concern we've been unable to observe in the real world" when in fact it means "people just haven't reported this definite bug yet".

**Comment by javiercn** [comment]
Closing since we don't believe this issue is relevant to any supported version of the product and the code has suffered dramatic changes over the years.
